system/CLI/Commands.php

```diff
@@ -146,7 +146,7 @@ public function verifyCommand(string $command, array $commands): bool
 
         $message = lang('CLI.commandNotFound', [$command]);
 
-        if ($alternatives = $this->getCommandAlternatives($command, $commands)) {
+        if (($alternatives = $this->getCommandAlternatives($command, $commands)) !== []) {
```

Comment:
Can we remove the assignment outside the if?
Also, is there a rector for that?

Reply:
yes, there is no rector yet for that iirc

Reply:
For my personal preference I use assign conditionals if the can be reordered Symfony-style:
```suggestion
        if ([] !== $alternatives = $this->getCommandAlternatives($command, $commands)) {
```
I won't try to impose that on our style guide, but to me it is way more readable and frequently saves an intermediary variable that will only be used in the comparison.

Reply:
@kenjis @MGatner do we use yoda style?

Reply:
I thought we don't use yoda style, but the php-cs-fixer rules permit only for identical (`===`, `!==`).
https://github.com/CodeIgniter/coding-standard/blob/cdd3a82d500325611075fc21d3fe261a69b26aec/src/CodeIgniter4.php#L639C14-L644

I think the following is fine:
```php
$alternatives = $this->getCommandAlternatives($command, $commands);
if ($alternatives !== []) {
```

Reply:
That was probably from my previous personal preference 😅 Again, really not trying to make a stand on this, open to team input.

Reply:
@paulbalandan Why do we permit yoda style only for identical?

Reply:
No, we do not. As far as I recall, @MGatner suggests using yoda for these types of situations.

Reply:
moved to previous assign.